Tests/UnitTests/Diagnostics/DiagnosticsFileHandlerTests.swift

```diff
@@ -0,0 +1,210 @@
+        await self.handler.appendEvent(diagnosticsEvent: Self.sampleEvent())
+
+        var entries = await self.handler.getEntries()
+        expect(entries.count).to(equal(4))
+
+        await self.handler.cleanSentDiagnostics(diagnosticsSentCount: 2)
+        entries = await self.handler.getEntries()
+        expect(entries.count).to(equal(2))
+
+        await self.handler.cleanSentDiagnostics(diagnosticsSentCount: 2)
+        entries = await self.handler.getEntries()
+        expect(entries.count).to(equal(0))
+    }
+
+    // MARK: - getEntries
+
+    func testGetEntries() async throws {
+        let line1 =
+            "{\"properties\":{\"key\":\"value\"}," +
+            "\"timestamp\":\"2024-04-04T12:55:59Z\"," +
+            "\"name\":\"HTTP_REQUEST_PERFORMED\"," +
+            "\"type\":\"event\"," +
+            "\"version\":1}"
+        let line2 =
+            "{\"properties\":{\"key\":\"value\"}," +
+            "\"timestamp\":\"2024-04-04T13:55:59Z\"," +
+            "\"name\":\"HTTP_REQUEST_PERFORMED\"," +
+            "\"type\":\"event\"," +
+            "\"version\":1}"
+
+        await self.fileHandler.append(line: line1)
+        await self.fileHandler.append(line: line2)
+
+        let content1 = DiagnosticsEvent(name: "HTTP_REQUEST_PERFORMED",
+                                        properties: ["key": AnyEncodable("value")],
+                                        timestamp: Date(millisecondsSince1970: 1712235359000))
+
+        let content2 = DiagnosticsEvent(name: "HTTP_REQUEST_PERFORMED",
+                                        properties: ["key": AnyEncodable("value")],
+                                        timestamp: Date(millisecondsSince1970: 1712238959000))
+
+        let entries = await self.handler.getEntries()
+        expect(entries[0]).to(equal(content1))
+        expect(entries[1]).to(equal(content2))
+    }
+
+    // MARK: - emptyFile
+
+    func testEmptyFile() async throws {
+        let line1 =
+            "{\"properties\":{\"key\":\"value\"}," +
+            "\"timestamp\":\"2024-04-04T12:55:59Z\"," +
+            "\"name\":\"HTTP_REQUEST_PERFORMED\"," +
+            "\"type\":\"event\"," +
+            "\"version\":1}"
+        let line2 =
+            "{\"properties\":{\"key\":\"value\"}," +
+            "\"timestamp\":\"2024-04-04T13:55:59Z\"," +
+            "\"name\":\"HTTP_REQUEST_PERFORMED\"," +
+            "\"type\":\"event\"," +
+            "\"version\":1}"
+
+        await self.fileHandler.append(line: line1)
+        await self.fileHandler.append(line: line2)
+
+        var data = try await self.fileHandler.readFile()
+        expect(data).toNot(beEmpty())
+
+        await self.handler.emptyDiagnosticsFile()
+
+        data = try await self.fileHandler.readFile()
+        expect(data).to(beEmpty())
+    }
+
+}
+
+@available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+class ModernDiagnosticsFileHandlerTests: BaseDiagnosticsFileHandlerTests {
```

Comment:
Interesting name 😅. Maybe `iOS15DiagnosticsFileHandlerTests`? I know it affects other systems, but usually we follow iOS I think. This might not be needed if we just drop support for diagnostics in iOS < 15